### PR TITLE
Add paper citation information to README

### PR DESCRIPTION
**Pull request by tgamblin**
This adds citation info and a link to our Archspec paper to the `README.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +49,19 @@ $ tox
   congratulations :)
 ```
 
+## Citing Archspec
+
+If you are referencing Archspec in a publication, please cite the following
```
Comment (boegel) [inline README.md:54]: `Archspec` -> `archspec`? (same in title above)

**Review by boegel — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (tgamblin) [reply README.md:54]: fixed

**Review by alalazo — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (boegel) [reply README.md:54]: It still looks the same to me? 🤔

Comment (tgamblin) [reply README.md]: oh I thought you meant the title and missed this.  Title is fixed.  This is also fixed now.

**Review by alalazo — APPROVED** [review]

**Review by boegel — APPROVED** [review]

**Pull request merged**